frameworks/FSharp/giraffe/src/App/StatefullRendering.fs

```diff
@@ -0,0 +1,60 @@
+﻿namespace App
+module rec StetefullRendering =
+
+    open System.Text
+    open Giraffe.GiraffeViewEngine
+    open System.Net
+    open System.IO
+
+    let inline private add (str:string) (target: StreamWriter) =
+        target.Write str
+        target
+
+    let inline private add' (str:string) (target: StreamWriter) =
+        target.Write str
+
+    let private closingBracket = ">"
+
+    let private writeStartElement target elemName (attributes : XmlAttribute array) : unit =
+
+        match attributes with
+        | [||] -> 
+            do target

```

Comment:
why are the so many explicit "do" statements used in this module? is there an error if not explicitly included? usually "do" is only required for iteration body statements and constructor initialization code?